src/librustdoc/doctest.rs

```diff
@@ -361,7 +361,7 @@ fn run_test(
     for debugging_option_str in &options.debugging_opts_strs {
         compiler.arg("-Z").arg(&debugging_option_str);
     }
-    if no_run && !compile_fail {
+    if no_run && !compile_fail && options.persist_doctests.is_none() {
```

Comment:
This may cause some minor divergence in behavior depending on whether `--persist-doctests` is run, for the same reason the `compile_fail` clause was added earlier. I don't have any perspective whether that's ok.